bin/import_ts.py

```diff
@@ -0,0 +1,103 @@
+
+from util.downloader import maybe_download
+from util.text import validate_label
+
+FIELDNAMES = ['wav_filename', 'wav_filesize', 'transcript']
+MAX_SECS = 10
+ARCHIVE_NAME = '2018-10-03_fr_FR'
+ARCHIVE_DIR_NAME = 'ts_' + ARCHIVE_NAME
+ARCHIVE_URL = 'https://s3.eu-west-3.amazonaws.com/audiocorp/releases/' + ARCHIVE_NAME + '.zip'
+
+
+def _download_and_preprocess_data(target_dir):
+    # Making path absolute
+    target_dir = path.abspath(target_dir)
+    # Conditionally download data
+    archive_path = maybe_download('ts_' + ARCHIVE_NAME + '.zip', target_dir, ARCHIVE_URL)
+    # Conditionally extract common voice data
+    _maybe_extract(target_dir, ARCHIVE_DIR_NAME, archive_path)
+    # Conditionally convert common voice CSV files and mp3 data to DeepSpeech CSVs and wav
+    _maybe_convert_sets(target_dir, ARCHIVE_DIR_NAME)
+
+def _maybe_extract(target_dir, extracted_data, archive_path):
+    # If target_dir/extracted_data does not exist, extract archive in target_dir
+    extracted_path = path.join(target_dir, extracted_data)
+    if not path.exists(extracted_path):
+        print('No directory "%s" - extracting archive...' % extracted_path)
+        if not os.path.isdir(extracted_path):
+            os.mkdir(extracted_path)
+        with zipfile.ZipFile(archive_path) as zip_f:
+            zip_f.extractall(extracted_path)
+    else:
+        print('Found directory "%s" - not extracting it from archive.' % archive_path)
+
+def _maybe_convert_sets(target_dir, extracted_data):
+    extracted_dir = path.join(target_dir, extracted_data)
+    # override existing CSV with normalized one
+    target_csv_template = os.path.join(target_dir, 'ts_' + ARCHIVE_NAME + '_{}.csv')
+    if os.path.isfile(target_csv_template):
+        return
+    path_to_original_csv = os.path.join(extracted_dir, 'data.csv')
+    with open(path_to_original_csv) as csv_f:
+        data = [
+            d for d in csv.DictReader(csv_f, delimiter=',')
+            if float(d['duration']) <= MAX_SECS
+        ]
+    with open(target_csv_template.format('train'), 'w') as train_csv_file:  # 80%
+        with open(target_csv_template.format('dev'), 'w') as dev_csv_file:  # 10%
+            with open(target_csv_template.format('test'), 'w') as test_csv_file:  # 10%
+                train_writer = csv.DictWriter(train_csv_file, fieldnames=FIELDNAMES)
+                train_writer.writeheader()
+                dev_writer = csv.DictWriter(dev_csv_file, fieldnames=FIELDNAMES)
+                dev_writer.writeheader()
+                test_writer = csv.DictWriter(test_csv_file, fieldnames=FIELDNAMES)
+                test_writer.writeheader()
+
+                for i, item in enumerate(data):
+                    transcript = validate_label(cleanup_transcript(item['text']))
+                    if not transcript:
+                        continue
+                    wav_filename = os.path.join(target_dir, extracted_data, item['path'])
+                    i_mod = i % 10
```

Comment:
Nice trick, but is the order of the original file random?

If not this will introduce correlations in the train, dev, and test sets. For example if the original file was ordered by user, then a single user could land in train, dev, and test.

Reply:
`data.csv` is ordered according to https://github.com/nicolaspanel/TrainingSpeech/blob/master/sources.json.

Since TrainingSpeech contains only a very few distinct speakers (4 right now, see https://github.com/nicolaspanel/TrainingSpeech#last-releases--download) the correlation will exists no matter how we split.

Reply:
BTW, the purpose of https://github.com/nicolaspanel/TrainingSpeech is to provide additional training data (ie not dev / test data) since speeches are extracted from audiobooks and therefore may not represent modern/common way of speaking

Reply:
@kdavis-mozilla 
I'm not saying we should drop the train/dev/test split. It is of course needed if your are using TrainingSpeech as your **only** datasource.

But, since TrainingSpeech samples are extracted from audiobooks and from a very few number of speakers, the performance on dev / test set will not represent real life performance such as noisy environments, familiar language, strong accents, etc.
A better approach would be to use dev / test sets that represent real life distributions. common-voice dev / test sets for example.

Therefore, IMOO, it is not the big deal if some (unwanted) correlations exists between train, dev et test.

However, I you prefer another approach  just tell me which one to use and I will send the changes
Best regards

Reply:
I think with the size of this data set we can't get away from unwanted correlations. So we'll just have to live with them for now ¯\\_(ツ)_/¯

That said, when combining this data set with others to train on, e.g. this training set is added to a second training set, having a separate test set for this data set will allow us to get a feel for how the model trained on the total training set performs on this data set. (Of course due to the correlations mentioned above this will not be perfect.)

Reply:
I should also say "real life performance" is use case specific. Someone may decide to use the model to transcribe audio books, then this data set reflects "real life performance" perfectly. So we should keep an open mind in this regard.